lib/net/modules/dns.toit

```diff
@@ -28,17 +26,38 @@ Look up a domain name and return an A or AAAA record.
 If given a numeric address like 127.0.0.1 it merely parses
   the numbers without a network round trip.
 
-By default the server is "8.8.8.8" which is the Google DNS
-  service.
+By default the server is determined by the network interface.  The fallback
+  servers if none are configured are the Google and Cloudflare DNS services.
 */
 dns_lookup -> net.IpAddress
     host/string
-    --server/string="8.8.8.8"
+    --server/string?=null
+    --client/DnsClient?=null
     --timeout/Duration=DNS_DEFAULT_TIMEOUT
     --accept_ipv4/bool=true
     --accept_ipv6/bool=false:
-  q := DnsQuery_ host --accept_ipv4=accept_ipv4 --accept_ipv6=accept_ipv6
-  return q.get --server=server --timeout=timeout
+  if server and client: throw "INVALID_ARGUMENT"
+  if not client:
+    if not server:
+      client = default_client
+    else:
+      client = ALL_CLIENTS_.get server --init=:
+        DnsClient [server]
+  return client.get host --accept_ipv4=accept_ipv4 --accept_ipv6=accept_ipv6 --timeout=timeout
+
+DEFAULT_CLIENT ::= DnsClient [
+    "8.8.8.8",  // Google.
+    "1.1.1.1",  // Cloudflare.
+    "8.8.4.4",  // Google.
+    "1.0.0.1",  // Cloudflare.
+]
+
+// A map from IP addresses (in string form) to DNS clients.
+// If a DNS client has multiple servers it can query then they
+// are separated by slash (/) in the key.
+ALL_CLIENTS_ ::= {DEFAULT_CLIENT.servers_.join "/": DEFAULT_CLIENT}
```

Comment:
Since we seem to pick servers at random, wouldn't it make sense to sort the sites?
If they were tried in order, then sorting would change semantics.